File: mesop/examples/web_component/svg_icon/svg_icon_component.js
import {
    LitElement,
    svg,
    html,
  } from 'https://cdn.jsdelivr.net/gh/lit/dist@3/core/lit-core.min.js';

export class SVGIconComponent extends LitElement {
    static properties = {
      svg: { type: String }
    };

    constructor() {
      super();
    }

    render() {
      const span = document.createElement('span');
      span.innerHTML = this.svg;
      return html`${span}`;
    }
  
  }
  
customElements.define('svg-icon', SVGIconComponent);

File: mesop/examples/web_component/svg_icon/svg_icon_component.py
import mesop as me


@me.web_component(path="./svg_icon_component.js")
def svg_icon(
  *,
  svg: str | None = None,
  key: str | None = None,
):
  return me.insert_web_component(
    name="svg-icon",
    key=key,
    properties={"svg": svg},
  )
